SimulationAgent/DeviceReplay/DeviceReplayActor.cs

```diff
@@ -0,0 +1,212 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.DataStructures;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Diagnostics;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Models;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.SimulationAgent.DeviceConnection;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.SimulationAgent.DeviceTelemetry;
+using static Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Models.DeviceModel;
+
+namespace Microsoft.Azure.IoTSolutions.DeviceSimulation.SimulationAgent.DeviceReplay
+{
+    public interface IDeviceReplayActor
+    {
+        void Init(
+            ISimulationContext simulationContext,
+            string deviceId,
+            DeviceModel deviceModel,
+            IDeviceConnectionActor context);
+
+        bool HasWorkToDo();
+        Task RunAsync();
+        void Stop();
+    }
+
+    public class DeviceReplayActor : IDeviceReplayActor
+    {
+        private enum ActorStatus
+        {
+            None,
+            ReadLine,
+            LineReady,
+            Stopped,
+            FileEnd,
+            Restart
+        }
+
+        private readonly ILogger log;
+        private readonly IActorsLogger actorLogger;
+        private ISimulationContext simulationContext;
+        private DeviceModel deviceModel;
+        private readonly IInstance instance;
+
+        private ActorStatus status;
+        private string deviceId;
+        private string currentLine;
+        private StreamReader streamReader;
+        private FileStream fileStream;
+        private long whenToRun;
+        private long prevInterval;
+        private IDeviceConnectionActor deviceContext;
+        private DeviceModelMessageSchema emptySchema;
```

Comment:
curious, why is this called emptySchema ? Should we rename this to replayFileSchema or something similar ?

Reply:
I just called it empty since it is a schema without any parameters and is therefore "empty". I'm not overly opinionated about the name so I can go either way.